portfolio/src/main/java/com/google/sps/servlets/MarkerServlet.java

```diff
@@ -0,0 +1,63 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.sps.Marker;
+import java.io.IOException;
+import java.util.ArrayList;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Handles fetching and saving markers data. */
+@WebServlet("/markers")
+public class MarkerServlet extends HttpServlet {
+
+  DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+  /** Responds with a JSON array containing marker data. */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    ArrayList<Marker> markers = new ArrayList<>();
+    Query query = new Query("Marker");
+    PreparedQuery results = datastore.prepare(query);
+
+    for (Entity entity : results.asIterable()) {
+      double lat = (double) entity.getProperty("lat");
```

Comment:
Nit: I believe Java has implicit casting. So I think it is easy to read if you are assigning the Object into a new variable in the same line without explicit casting. Not even sure if implicit casting with the getProperty method. Please let me know as a reply to this comment.

Reply:
I looked into this and Java does have implicit casting into types with a larger range (I didn't know this!). I believe this means I cannot implicitly cast into the Object type though. I tried it in my code and got the error "incompatible types: java.lang.Object cannot be converted to double".

the documentation on [getProperty](https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/PropertyContainer.html#getProperty-java.lang.String-) is sort of vague on what type of Object it returns and the [setProperty](https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/PropertyContainer.html#setProperty-java.lang.String-java.lang.Object-) documentation names some conversions made but not all
So I looked at code examples [here](https://www.programcreek.com/java-api-examples/?class=com.google.appengine.api.datastore.Entity&method=getProperty) and it appears that all calls to getProperties are explicitly cast into something else

Reply:
Thanks for verifying this for me! LGTM